### ISSUE TITLE
Players can see map through walls

### ISSUE THREAD

**Issue by jontrost** [cyruzzo/AboveVTT#1293]
First of all, thanks in advance for all the hard work put into this project. I've been a Patreon supporter for a while now because as a software engineer myself, I understand the difficulty of making an application like this. The work that goes into this project does not go unappreciated.

Now for the issue:
When a player joins the VTT, they cannot see tokens on the opposite side of a wall from their token. This is the expected behavior, but I noticed that the map on the other side of the wall is still visible to the player. If the "selected token vision" option is selected, I see exactly what I would expect. The player cannot see the map or the tokens on the other side of a wall. As soon as the player clicks on anything other than their token, they can see the entire map again even though sections are completely blocked by walls. The solutions I see are either making it so that players can be locked into their token view at all times, or making it so that a player who doesn't have their token currently selected cannot see any portions of the map that are on the opposite side of a wall from their token. These are essentially the same, but also slightly different.

Here are some screenshots for reference. The player's view with the token selected is what I would expect to be able to restrict a player's view to. AboveVTT clearly knows what the player should be able to see before they select their token since it can correctly hide the tokens on opposite sides of a wall, but it should also hide the room itself.

**DM View**
![dm-view](https://github.com/cyruzzo/AboveVTT/assets/35822999/b8c1f170-08fc-41af-8033-90a66de1a782)

**Player View (unselected token)**
![unselected-view](https://github.com/cyruzzo/AboveVTT/assets/35822999/c4ae68dd-b087-43be-ad1a-911d7062c7b1)

**Player View (selected token)**
![selected-view](https://github.com/cyruzzo/AboveVTT/assets/35822999/35a5214a-569d-47f2-a7a6-e9bed1d1f332)

**Comment by Azmoria** [comment]
If you set the darkness filter to 100% on the scene it should resolve the issue you're having. You can then light the scene with either with light using the draw tool or light on hidden tokens. 

Lots of people run into this and I'm brainstorming ways to move options around for vision/light so it might be more intuitive.

The map staying visible to players is generally used when they know the layout of an area. It also allows for fog to be used with darkness to leave a trail behind players - although this is currently a manual process which isn't ideal.

**Comment by jontrost** [comment]
Thanks for the response, that should work I think. I tried messing around with the darkness filter to solve the issue, but I assumed the Universal VTT format would also preserve the lighting from when I created the map, and I think I was incorrect in that assumption considering even the rooms that the tokens are in were dark.

I'll still suggest that a way to lock a players vision to only their specific token would be a fantastic feature. Would it be better to close this issue and open a new one stating exactly that or just leave this ticket as-is?

**Comment by Azmoria** [comment]
Unless another token has shared vision enabled players should be locked to their own token vision. They shouldn't see any difference with the token selected or not at 100% darkness if there is no shared vision. Below 100% it would still be their own vision for revealing other tokens - if another player stepped around a corner for example their token would be hidden to them.

The only exception to this is if they don't have their pc token placed then they will see all players that do have tokens on the map - generally used for streaming or a DM player tab.

**Comment by Azmoria** [comment]
As to a new issue or not you can leave this one up - I'm going to mark it with a potential workaround tag in case others happen to see it. At least until we have better QoL for this stuff.

**Comment by jontrost** [comment]
I just checked and I can confirm that the program the file was exported from did _not_ include the lighting, so I don't think there's any bug here. It sounds like the behavior I was expecting should be achievable with the workaround you mentioned. Thanks again for the responses. Feel free to close this ticket if that's appropriate, but I'll leave it open for now in case that's desired.

**Comment by Azmoria** [comment]
Doing some clean up of older issues - closing this as I believe we found the issue at the time/it's resolved. I'll add a new post for the request to lock vision.
